### PR TITLE
Document protocol changes requirements

### PR DESCRIPTION
**Pull request by janherich**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

Documenting protocol changes ADR, capturing all important decisions and rules made recently.

status: ready

### CODE REVIEW

File: doc/decisions/0011-protocol-version-compatibility.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# 0011. Protocol compatibility between App versions
+
+| Date | Tags |
+|---|---|
+| Fri Oct 05  | protocol, messagner |
```
Comment (pedropombeiro) [inline doc/decisions/0011-protocol-version-compatibility.md]: typo: `messenger`

File: doc/decisions/0011-protocol-version-compatibility.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## Status
+
+draft
+
+## Context
+
+This documents enumerates the rules we have to adhere to when doing any protocol modifications and summarizes best practises to do so.
```
Comment (pedropombeiro) [inline doc/decisions/0011-protocol-version-compatibility.md]: typo: `document`

File: doc/decisions/0011-protocol-version-compatibility.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+Rules for making any changes related to message payload (transit fields defined in `status-im.transport.message.transit` ns):
+
+- Every new change must be backward compatible back to released App version `0.9.20` and higher. That means that whenever for example
+  new feature (like message replies recently) is added, we can't of course do nothing with the fact that older versions can't properly
+  display new feature, but we must ensure that at least functionality existing in the older version is not affected at all.
+  So in the example above, even if the older version can't display the fact that some message is reply to other message, it still has to
+  be able to receive and display it as a standalone message.
+  
+- This means we **can't** under no circumstances increment the tag of the existing message type handler (like `c4` -> `c5`) because older
```
Comment (pedropombeiro) [inline doc/decisions/0011-protocol-version-compatibility.md]: it would be clearer with `under any circumstances`

**Review by pedropombeiro — APPROVED** [review]
Good stuff! Thanks!

**Comment by pedropombeiro** [comment]
We'll probably want to have a document outlining the procedure to follow when a version with an incompatible protocol is released (i.e. in the case of a nightly, do we account for the bad release in the future protocol changes? for how long?). I guess this can be done in the Upgrade Policy document.

File: doc/decisions/0011-protocol-version-compatibility.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+- You can join public channel like `#status` with PR build, but never post anything there unless you are sure that everything is working
+  flawless in 1-1 and public channels dedicated to testing, including compatibility between PR build and older versions.
+  Of course as a last step, testing in busy public channel like `#status` is advised, because if despite all the effort, something goes
+  bad, it's better when it's discovered when "dangerous" version is in possesion of just one tester and not already released to the wild,
+  even if only as a nightly build for eager desktop users.
+  
+## Consequences
+
+Greater understanding and clarity for requirements about protocol changes.
```
Comment (oskarth) [inline doc/decisions/0011-protocol-version-compatibility.md:56]: I feel like some of the stuff in decision are actually consequences, maybe

File: doc/decisions/0011-protocol-version-compatibility.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+| Fri Oct 05  | protocol, messenger |
+
+## Status
+
+draft
+
+## Context
+
+This document enumerates the rules we have to adhere to when doing any protocol modifications and summarizes best practises to do so.
+In the light of recent issues (code landed in develop branch, producing "invisible" messages for previous app versions), it's best
```
Comment (oskarth) [inline doc/decisions/0011-protocol-version-compatibility.md:14]: People won't know this recent issue, perhaps we can rewrite this as a form of "XX happened which lead to YY problem"

File: doc/decisions/0011-protocol-version-compatibility.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+to capture it in the ADR for future reference.
+
+## Decision
```
Comment (oskarth) [inline doc/decisions/0011-protocol-version-compatibility.md:17]: Not sure how, but perhaps we can format this a better nicer so it is easier to read/clearer? Feels a bit wordy right now

**Review by oskarth — COMMENTED** [review]
Thanks for starting this! Left some comments

File: doc/decisions/0011-protocol-version-compatibility.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+Rules for making any changes related to message payload (transit fields defined in `status-im.transport.message.transit` ns):
+
+- Every new change must be backward compatible back to released App version `0.9.20` and higher. That means that whenever for example
+  new feature (like message replies recently) is added, we can't of course do nothing with the fact that older versions can't properly
+  display new feature, but we must ensure that at least functionality existing in the older version is not affected at all.
+  So in the example above, even if the older version can't display the fact that some message is reply to other message, it still has to
+  be able to receive and display it as a standalone message.
+  
+- This means we **can't** under any circumstances increment the tag of the existing message type handler (like `c4` -> `c5`) because older
```
Comment (yenda) [inline doc/decisions/0011-protocol-version-compatibility.md:27]: With PSS old version won't be forward compatible so we'll ditch everything. We should plan ahead to make sure we clean up and think of the future as much as possible, as it will be the last occasion

**Comment by oskarth** [comment]
@yenda would you mind taking over this PR since @janherich left? Or we can close it

**Comment by rasom** [comment]
No activity here, let's close PR
